### PR TITLE
Only scroll/focus a FileTreeNode when necessary

### PR DESCRIPTION
**Pull request by kumar303**
This fixes https://github.com/mozilla/addons-code-manager/issues/699 by keeping track of focused nodes and only scrolling when they first come into focus.

### CODE REVIEW

File: src/components/FileTree/index.tsx [hunk 1]
```diff
@@ -78,7 +78,9 @@ export class FileTreeBase extends React.Component<Props> {
     const { onSelect, version } = this.props;
 
     if (version) {
-      return <FileTreeNode {...props} onSelect={onSelect} version={version} />;
+      return (
+        <FileTreeNode {...props} onSelect={onSelect} versionId={version.id} />
```
Comment (willdurand) [inline src/components/FileTree/index.tsx]: This and the change in `FileTreeNode` go against https://github.com/mozilla/addons-code-manager/pull/400#discussion_r267058168

File: src/test-helpers.tsx [hunk 2]
```diff
@@ -650,3 +651,12 @@ export const createFakeRef = (
     },
   };
 };
+
+export const createStoreWithVersion = (
```
Comment (willdurand) [inline src/test-helpers.tsx]: Should we file an issue to use this helper function everywhere else?

File: src/components/FileTree/index.tsx (hunk 1 above)
Comment (kumar303) [reply src/components/FileTree/index.tsx]: Yes, it does go against it which is why I filed https://github.com/mozilla/addons-code-manager/issues/754 to fix it. There's a todo comment about it `mapStateToProps()`.

The problem here is that we can't rely on a passed `version` object *and* be guaranteed a re-render when `version.focusedPath` changes. We need to map the object to state to guarantee an update.

File: src/test-helpers.tsx (hunk 2 above)
Comment (kumar303) [reply src/test-helpers.tsx]: Yeah, good idea. I did a quick search and it can be used in a lot of places.

**Comment by kumar303** [comment]
> It took me a while to understand how a "focused path" would fix this issue given that the selected path was always visible and therefore looked "focused" enough to me, but I got it.

Well, the selected path was not always visible. In the case where scrolling was needed, the path node was off the screen in the panel overflow; it was invisible.

> 
> I think something mentioning `scroll` (like `lastScrolledPath`, maybe) would make things more obvious given that "focus" is often referred to things visible, but I am not a native speaker 🤷‍♂

Hmm, *scrolling into view* is more of an implementation of making the path node visible where it was previously invisible. I can see how *focused* could be confusing, though, since it implies a real DOM focus. I changed the names to use the word *visible* instead.

Thanks for the review.

**Comment by kumar303** [comment]
The Snyk report won't load for me (times out). Let's see what happens.

**Pull request merged**